src/main/kotlin/com/avail/interpreter/levelTwo/operation/L2_RUN_INFALLIBLE_PRIMITIVE.kt

```diff
@@ -0,0 +1,306 @@
+ *
+ * * Neither the name of the copyright holder nor the names of the contributors
+ *   may be used to endorse or promote products derived from this software
+ *   without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.avail.interpreter.levelTwo.operation
+
+import com.avail.descriptor.types.A_Type
+import com.avail.interpreter.Primitive
+import com.avail.interpreter.levelTwo.*
+import com.avail.interpreter.levelTwo.L2Operation.HiddenVariable.*
+import com.avail.interpreter.levelTwo.operand.*
+import com.avail.optimizer.L2Generator
+import com.avail.optimizer.RegisterSet
+import com.avail.optimizer.jvm.JVMTranslator
+import org.objectweb.asm.MethodVisitor
+import java.util.*
+import java.util.function.Consumer
+
+/**
+ * Execute a primitive with the provided arguments, writing the result into
+ * the specified register.  The primitive must not fail.  Don't check the result
+ * type, since the VM has already guaranteed it is correct.
+ *
+ *
+ *
+ * Unlike for [L2_INVOKE] and related operations, we do not provide
+ * the calling continuation here.  That's because by inlining the primitive
+ * attempt we have avoided (or at worst postponed) construction of the
+ * continuation that reifies the current function execution.  This is a Good
+ * Thing, performance-wise.
+ *
+ *
+ * @author Mark van Gulik &lt;mark@availlang.org&gt;
+ * @author Todd L Smith &lt;todd@availlang.org&gt;
+ *
+ * @constructor
+ * Construct an `L2_RUN_INFALLIBLE_PRIMITIVE`.
+ */
+abstract class L2_RUN_INFALLIBLE_PRIMITIVE private constructor()
+	: L2Operation(
+		L2OperandType.CONSTANT.`is`("raw function"),  // Used for inlining/reoptimization.
+		L2OperandType.PRIMITIVE.`is`("primitive to run"),
+		L2OperandType.READ_BOXED_VECTOR.`is`("arguments"),
+		L2OperandType.WRITE_BOXED.`is`("primitive result"))
+{
+	/** The subclass for primitives that have no global dependency.  */
+	@WritesHiddenVariable(LATEST_RETURN_VALUE::class)
+	private class L2_RUN_INFALLIBLE_PRIMITIVE_no_dependency
+		: L2_RUN_INFALLIBLE_PRIMITIVE()
+
+	/** The subclass for primitives that have global read dependency.  */
+	@ReadsHiddenVariable(theValue = arrayOf(GLOBAL_STATE::class))
```

Comment:
Rewrite "(theValue = arrayOf(...))" to "([...])".
Same for "(value = arrayOf(...))" on the next line, and subsequent occurrences in this file.

Reply:
you have to keep:
`value =[...]`
for some reason